sycl/include/sycl/detail/kernel_desc.hpp

```diff
@@ -277,6 +277,52 @@ template <typename KernelNameType> constexpr bool hasSpecialCaptures() {
   }
   return FoundSpecialCapture;
 }
+inline namespace compile_time_kernel_info_v1 {
+
+// This is being passed across ABI boundary, so we don't use std::string_view,
+// at least for as long as we support user apps built with GNU libstdc++'s
+// pre-C++11 ABI.
+struct CompileTimeKernelInfoTy {
+  const char *const Name = nullptr;
+  const unsigned NumParams = 0;
+  const bool IsESIMD = false;
+  const char *const FileName = "";
+  const char *const FunctionName = "";
+  const unsigned LineNumber = 0;
+  const unsigned ColumnNumber = 0;
+  const int64_t KernelSize = 0;
+  kernel_param_desc_t (*const ParamDescGetter)(int) = nullptr;
+  const bool HasSpecialCaptures = false;
```

Comment:
```suggestion
  const bool HasSpecialCaptures = true;
```

might be a better choice for now, to align with https://github.com/intel/llvm/blob/c7ff66105a7b7dfa2c8a3544f0b132e39ace7594/sycl/source/detail/scheduler/commands.hpp#L627-L639